src/wasm/asm-wasm-builder.cc

```diff
@@ -117,9 +121,50 @@ class AsmWasmBuilderImpl : public AstVisitor {
     }
   }
 
-  void VisitContinueStatement(ContinueStatement* stmt) {}
+  void VisitContinueStatement(ContinueStatement* stmt) {
+    DCHECK(in_function_);
+    int i = static_cast<int>(breakable_blocks_.size()) - 1;
+    int jmp_ht = 0;
+    for (; i >= 0; i--) {
+      auto elem = breakable_blocks_.at(i);
+      if (elem.first == stmt->target()) {
+        DCHECK(elem.second);
+        current_function_builder_->AppendCode(kExprBr, false);
+        current_function_builder_->AppendCode(jmp_ht, false);
+        current_function_builder_->AppendCode(kExprNop, false);
+        break;
+      } else if (elem.second) {
+        jmp_ht += 2;
+      } else {
+        jmp_ht += 1;
+      }
+    }
+    DCHECK(i >= 0);
+  }
 
-  void VisitBreakStatement(BreakStatement* stmt) {}
+  void VisitBreakStatement(BreakStatement* stmt) {
+    DCHECK(in_function_);
+    int i = static_cast<int>(breakable_blocks_.size()) - 1;
+    int jmp_ht = 0;
+    for (; i >= 0; i--) {
+      auto elem = breakable_blocks_.at(i);
+      if (elem.first == stmt->target()) {
+        current_function_builder_->AppendCode(kExprBr, false);
+        if (elem.second) {
+          current_function_builder_->AppendCode(jmp_ht + 1, false);
```

Comment:
Can you extract the target of the jump (i.e. jmp_ht or jmp_ht + 1) out to a local variable?